### PR TITLE
(build) Target net46

### PR DESCRIPTION
**Pull request by mholo65**
Let's see how this goes... Was unable to build on my Linux box, Mono 5 segfaulted as usual 😄

### CODE REVIEW

File: src/Cake/Cake.csproj [hunk 1]
```diff
@@ -27,16 +27,14 @@
     <PackageReference Include="Autofac" Version="4.0.0-rc3-280" />
     <PackageReference Include="System.Diagnostics.FileVersionInfo" Version="4.3.0" />
     <PackageReference Include="System.Threading.Thread" Version="4.3.0" />
-    <PackageReference Include="Microsoft.NET.Sdk">
-      <Version>1.0.0-alpha-20161104-2</Version>
-      <PrivateAssets>All</PrivateAssets>
-    </PackageReference>
```
Comment (mholo65) [inline src/Cake/Cake.csproj]: It's not recommended to explicitly specify this

File: src/Cake.Common/Cake.Common.csproj [hunk 2]
```diff
@@ -32,14 +32,14 @@
...
+    <Reference Include="System" />
+    <Reference Include="System.IO.Compression" />
+    <Reference Include="System.IO.Compression.FileSystem" />
     <Reference Include="System.Net.Http" />
     <Reference Include="System.Runtime.Serialization" />
     <Reference Include="System.Runtime.Serialization.Json" />
-    <Reference Include="System.IO.Compression" />
-    <Reference Include="System.IO.Compression.FileSystem" />
-    <Reference Include="System" />
+    <Reference Include="System.Xml" />
```
Comment (mholo65) [inline src/Cake.Common/Cake.Common.csproj]: Sorry, just my OCD reordering these

File: build.cake [hunk 3]
```diff
@@ -183,15 +180,15 @@ Task("Copy-Files")
     CopyFileToDirectory("./LICENSE", parameters.Paths.Directories.ArtifactsBinNetCore);
 
     // Copy Cake.XML (since publish does not do this anymore)
-    CopyFileToDirectory("./src/Cake/bin/" + parameters.Configuration + "/net462/Cake.xml", parameters.Paths.Directories.ArtifactsBinFullFx);
-    CopyFileToDirectory("./src/Cake/bin/" + parameters.Configuration + "/netcoreapp1.0/netcoreapp1.0/Cake.xml", parameters.Paths.Directories.ArtifactsBinNetCore);
+    CopyFileToDirectory("./src/Cake/bin/" + parameters.Configuration + "/net46/Cake.xml", parameters.Paths.Directories.ArtifactsBinFullFx);
+    CopyFileToDirectory("./src/Cake/bin/" + parameters.Configuration + "/netcoreapp1.0/Cake.xml", parameters.Paths.Directories.ArtifactsBinNetCore);
```
Comment (mholo65) [inline build.cake]: Dunno why this was duplicated before. Might have to do with the stuff I removed from `Cake.csproj`

File: src/Cake.Core/Scripting/ScriptConventions.cs [hunk 4]
```diff
@@ -81,6 +81,9 @@ public IReadOnlyList<Assembly> GetDefaultAssemblies(DirectoryPath root)
             result.Add(typeof(System.Xml.XmlReader).GetTypeInfo().Assembly); // System.Xml
             result.Add(typeof(System.Xml.Linq.XDocument).GetTypeInfo().Assembly); // System.Xml.Linq
             result.Add(typeof(System.Data.DataTable).GetTypeInfo().Assembly); // System.Data
+
+            // This is just to please Roslyn when running under Mono. See issue https://github.com/dotnet/roslyn/issues/19364
+            result.Add(_loader.Load(new AssemblyName("System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"))); // System.Runtime
```
Comment (patriksvensson) [inline src/Cake.Core/Scripting/ScriptConventions.cs]: Should we add a Mono check here, or do we always need `System.Runtime`?

```
if(Type.GetType("Mono.Runtime") != null) {
  result.Add(_loader.Load(new AssemblyName("System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"))); // System.Runtime
}

Comment (mholo65) [inline src/Cake.Core/Scripting/ScriptConventions.cs]: @patriksvensson I'd say it's always safe to have `System.Runtime`. It's really just a facade referencing `mscorlib`, `System`, `System.ComponentModel.Composition` and `System.Core`.

Comment (devlead) [inline src/Cake.Core/Scripting/ScriptConventions.cs]: I'll do some testing but seems to work fine on Windows. So I'll say keep it in.

**Comment by mholo65** [comment]
@devlead @patriksvensson I have requested @Redth taking this for a spin also.

**Comment by devlead** [comment]
> @devlead @patriksvensson I have requested @Redth taking this for a spin also.

Excellent 👍

**Comment by devlead** [comment]
@mholo65  your changes have been merged, thanks for your contribution 👍

**Pull request merged**